pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java

```diff
@@ -46,6 +46,7 @@
     protected final PulsarClientImpl client;
 
     protected volatile ProducerImpl producer;
+    public static final String producerNamePadding = "####@@@@!!!!padding";
```

Comment:
Not really a big fan of having these random chars in the log :) 

Given that the current parsing logic is based on splitting at the `.`, I see 3 options here: 

 1. Currently we use `pulsar.repl.SOURCE` . Change into something like `pulsar.repl.SOURCE.-->.TARGET` so that the padding is used to indicate the data direction.
 2. Fix the parsing logic to expect a better format and do a 2 steps rollout 
 3. Just change the format and live with stats inconsistencies during the upgrade

Reply:
> 2. Fix the parsing logic to expect a better format and do a 2 steps rollout

👍

Reply:
made the change at producer which will be able to parse producer-name with delimiter and will make the next change of passing-remote-cluster-name in next release to avoid compatibility issue.